src/test/kotlin/org/jetbrains/research/ml/coding/assistant/tree/PsiTreeConverterTest.kt

```diff
@@ -20,29 +22,36 @@ class PsiTreeConverterTest : ParametrizedBaseTest(getResourcesRootPath(::PsiTree
 
     companion object {
         @JvmStatic
-        @Parameterized.Parameters(name = "{index}: ({0}, {1})")
+        @Parameterized.Parameters(name = "{index}-{2}: ({0}, {1})")
         // TODO: which tests cases should I have?
         fun getTestData(): List<Array<Any>> {
-            val files = getNestedFiles(getResourcesRootPath(::PsiTreeConverterTest))
+            val files = getInAndOutFilesMap(getResourcesRootPath(::PsiTreeConverterTest))
             val numberings = listOf(PreOrderNumbering, PostOrderNumbering)
-            return files.flatMap { f -> numberings.map { n -> arrayOf(f, n) } }.toList()
+            return files.flatMap { f -> numberings.map { n -> arrayOf(f.key, f.value, n) } }.toList()
         }
     }
 
     @JvmField
     @Parameterized.Parameter(0)
-    var inFile: File? = null
+    var inSourceFile: File? = null
 
     @JvmField
     @Parameterized.Parameter(1)
+    var outXmlFile: File? = null
+
+    @JvmField
+    @Parameterized.Parameter(2)
     var numbering: Numbering? = null
 
     @Test
     fun `converting PSI to GumTree tree test`() {
-        val inFilePsi = myFixture.configureByFile(inFile!!.name)
+        val inFilePsi = myFixture.configureByFile(inSourceFile!!.absolutePath.replace(testDataPath, ""))
         val inContext = ApplicationManager.getApplication().runReadAction<TreeContext> {
             PsiTreeConverter.convertTree(inFilePsi, numbering!!)
         }
         TestCase.assertTrue(inFilePsi.equalTreeStructure(inContext))
+        val expectedXml = outXmlFile!!.content
+        val actualXml = TreeIoUtils.toXml(inContext).toString().removeSuffix("\n")
+        TestCase.assertEquals(actualXml, expectedXml)
```

Comment:
do we need to have both structure and XML checks?

Reply:
I am not sure, but we should compare ids and we can check it in the `equalTreeStructure`.

Reply:
Maybe it's better to split up tests for id check and label check to not have several asserts in one test?